lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -219,6 +219,11 @@ static void addHLSLPasses(bool HLSLHighLevel, unsigned OptLevel, hlsl::HLSLExten
     MPM.add(createHLDeadFunctionEliminationPass());
   }
 
+  if (!NoOpt) {
+    MPM.add(createLoopRotatePass());
+    MPM.add(createDxilLoopUnrollPass());
```

Comment:
>MPM.add(createDxilLoopUnrollPass()); [](start = 4, length = 36)

Is one of these two pass responsible for honoring the [unroll] attribute? If so, it should be run even at opt level 0